railib/api.py

```diff
@@ -148,7 +148,7 @@ def delete_user(ctx: Context, user: str) -> dict:
 
 
 def get_engine(ctx: Context, engine: str) -> dict:
-    return _get_resource(ctx, PATH_ENGINE, name=engine, key="computes")
+    return _get_resource(ctx, PATH_ENGINE, name=engine, deleted_on="", key="computes")
```

Comment:
Sorry I merged it prematurely (by accident). I tried to explain the list behavior of /compute vs /database in [this](https://github.com/RelationalAI/relationalai-infra/pull/1832#issue-1053788214) PR description. Basically, the /compute endpoint returns both the deleted & non-deleted engines by default (i.e. without any filters) for compatibility reasons. If we'd change it to return only the non-deleted engines (which, I agree, it'd be better), it'd break the existing clients that rely on the default behavior to detect when a compute is provisioned/deleted.

After https://github.com/RelationalAI/relationalai-infra/pull/1832, a name can be shared among multiple computes (1 non-deleted & multiple deleted). By adding the empty `deleted_on` filter we make sure we search only for the 1 non-deleted compute which might be in one of the various provisioning states (requested, provisioning, provision failed, provisioned, etc.). So, instead of filtering on empty `deleted_on`, we could filter on multiple provisioning states or we could just filter on e.g. `PROVISIONED`. What do you think?